mirrord-layer/src/pod_api.rs

```diff
@@ -22,15 +22,38 @@ struct RuntimeData {
 }
 
 impl RuntimeData {
-    async fn from_k8s(client: Client, pod_name: &str, pod_namespace: &str) -> Self {
+    async fn from_k8s(
+        client: Client,
+        pod_name: &str,
+        pod_namespace: &str,
+        container_name: &Option<String>,
+    ) -> Self {
         let pods_api: Api<Pod> = Api::namespaced(client, pod_namespace);
         let pod = pods_api.get(pod_name).await.unwrap();
         let node_name = &pod.spec.unwrap().node_name;
         let container_statuses = pod.status.unwrap().container_statuses.unwrap();
-        let container_info = container_statuses
-            .first()
-            .unwrap()
-            .container_id
+        let container_info_map = container_statuses
```

Comment:
I think maybe just looping would be better, this is a bit of a nitpick but creating the hashmap, cloning data then just getting one key is overkill then to just check on the iteration.
Possibly https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find

Reply:
You're totally right, `find()` would be simpler and this is pretty convoluted :)